### DISCUSSION TITLE
integration with orm edgy

### DISCUSSION THREAD

**Discussion by YuriFontella** [dymmond/esmerald#255] [General]
Hello, 

I'm trying emerald and edgy, and I have this question about data return.
Using the orm function to return data in the route, data serialization works.
When I directly use the fetch_one function with a raw query, it is unable to send the data in the request.

![image](https://github.com/dymmond/esmerald/assets/4920619/fae7655e-7725-46d8-b026-a102e001cc03)

I don't know exactly if this is a problem, and whether it applies to edgy or emerald.
I may be doing something wrong, I appreciate any information.
Thanks.

**Comment by tarsil** [comment]
@YuriFontella hi. Thank you for this. The reason for this it's because the databasez returns a type SQLA Row and that it's not serializable 🙂

Does this make sense? The best way is by using the get() where you can filter by whatever you want or if you want to use the databasez like you are doing, you need to convert the result into a serializable format.

Edgy in the docs explains how to use the queries 🙂

**Comment by YuriFontella** [comment]
I understand that most database adapters do not explicitly return a list or a dict and that web frameworkers need this to send the data correctly in the request.

But I was looking for something like https://www.encode.io/databases without an orm behind it, it seems like an obsolete project.
And the query data that returns from the databases, works in fastapi serialization.

Creating a new list with the result seems a little tiring to me.
With the ORM everything works perfectly, you did a great job.

**Comment by tarsil** [comment]
Oh thank you very much @YuriFontella i do appreciate it. How is Esmerald so far? Well Databasez was forked from databases to serve other use cases and continue from there. It was necessary.

If you want to return a list of results with Esmerald you can change the return signature to be List[User] and return the users from the queryset and automatically Esmerald handles the response for you. 

The query data that returns from that is not pydantic oriented, it's simple python objects. If you are looking for something like that, you can use [Saffier](https://saffier.tarsild.io/saffier) instead. It's basically a more pure orm than edgy. Edgy leverages pydantic at the core basically.

I'm not sure if I'm answering you to be honest and If not, I apologize

**Reply by YuriFontella** [reply]
Everything is fine, thank you very much.
I will definitely use the esmerald in the next project.

Last question
Because databazes is a fork, does it also use sqlalchemy 1.4 or has it migrated to 2.0?

**Reply by tarsil** [reply]
No problem 🙂. The main reason for the fork was to move to 2.0 so yes, it's in 2.0 but then more things were added like support to SQL server and such.

I'm glad you chose Esmerald 🙂

**Reply by YuriFontella** [reply]
I just saw it in the motivation text... god 😱

**Reply by tarsil** [reply]
Is that good? Lol

**Reply by YuriFontella** [reply]
In closing... I'm just a web developer 😆

knows how to answer because what does fastapi do differently? Is that it can serialize and send the result of the query made by databasez.

![image](https://github.com/dymmond/esmerald/assets/4920619/358667ca-1e87-4eff-ae24-1eb30298112b)

**Reply by tarsil** [reply]
Hmm I can have a look at an improvement to Esmerald. Esmerald is more consistent in different patterns compared to FastAPI. They serve differently and in different ways.

Btw, Portuguese? (Brazilian?) If yes, so do I (Portugal) 🙂

**Reply by YuriFontella** [reply]
It would be great to have the facilities of fastapi with the performance of starlette and the advantages of esmerald (still exploring the documentation).

Legal, percebi pelo seu nome que é muito comum aqui no Brasil 🤣

**Reply by tarsil** [reply]
Sim, chamo me Tiago. Sou de Portugal 🙂. O Esmerald tem literalmente coisas que o FastAPI tem. Mais versatilidade, modularidade e coisas que o FastAPI não tem porque não é o caso de uso deles. O Esmerald corre também em cima do Starlette, daí a performance ser também alta. 

Em relação ao response, o Esmerald é mais controlado mas nas próximas versões vai se adicionar uma coisa nova que pode ajudar em praticamente todos os responses como aquele que mencionaste. 

No entanto não é um blocker 🙂.

Espero que gostes do Esmerald e tás sempre a vontade para contribuir caso aches que precisa 🙂

**Reply by YuriFontella** [reply]
Valeu Tiago, obrigado por suas contribuições. Já marquei o esmerald e o databasez para me notificar com novas releases.

**Reply by tarsil** [reply]
Espero que gostes do Esmerald e realmente tires proveito. Eu acho que vais gostar e vais usar muito 🙂
